README.md

```diff
@@ -0,0 +1,66 @@
+OLM Manifests
+=============
+
+Integration point for all the components of KubeVirt's OLM related manifests.
+The goal of this project is to make a KubeVirt package, viewable in OLM, such
+that a subscription to KubeVirt provides all of components of KubeVirt.
+
+# Cluster Requirements
+
+The [`registry`](registry/) directory makes use of OLM's bundle support that is
+provided only via the `grpc` type `CatalogSource`. You can see us create this
+`CatalogSource` in the provided [`playbook.yaml`](playbook.yaml). This means
+that you **must** have a relatively new **OpenShift 4.0** cluster.
+
+# Ansible Requirements
+
+If you want to run the playbook you will need a recent version of Ansible (`>=
```

Comment:
whoa... playbooks? what's this about. Can't all of this be managed by posting manifests using oc/kubectl tooling?

Reply:
You absolutely could use kubectl|oc. I only wrote the playbook because I was tired of running `sed "s|image: .*|image: docker.io/djzager/kubevirt-operators:sometag|" | kubectl (create|apply|delete) -f -` and a simple playbook was the easiest method available.

It's totally optional and can be removed, but I do believe that it makes iterating on this stuff easier.

Reply:
This sounds like you're looking for code to generate manifests using a template. We have to handle the same thing in kubevirt.

Take a look at the https://github.com/kubevirt/kubevirt/tree/master/tools/manifest-templator file. We use that to inject dynamic data into templated manifests from variables to do exactly what you're talking about. 

It should be a goal to remove dependency of ansible from the dev workflow

Reply:
Are you using manifests as a general reference to YAML files with one or more Kubernetes object specs inside or the components to an OLM CSV? If you are talking about the latter, I think updating this project to generate a KubeVirt CSV from all of the component projects is absolutely something we should do. In that respect I would need a little more guidance on how we proceed down that route (ie. do I include all the other projects as submodules in this one, grab them by github URL, or grab them out of a released image of KubeVirt|CDI|etc).

Writing a manifest-templator so that I can template out a Namespace, OperatorGroup, and CatalogSource for the sole purpose of deploying it into my development cluster strikes me as an odd use of time. If removing the dependency on Ansible is important, I would just make a single yaml file and go back to my `sed + kubectl` method.

> It should be a goal to remove dependency of ansible from the dev workflow

I guess that all of my working with Ansible has biased me somehow. Your saying that it should be a goal to remove this Ansible dependency makes me think there is more to this than just a preference. For my edification, would you be able to elaborate?

Reply:
> I guess that all of my working with Ansible has biased me somehow. Your saying that it should be a goal to remove this Ansible dependency makes me think there is more to this than just a preference. For my edification, would you be able to elaborate?

we've been moving toward a pattern where manifests are generated by code. Take https://github.com/kubevirt/kubevirt/pull/2062 for example. The kubevirt olm manifest originates from golang code.

Ideally this project venders the work kubevirt has done to generate manifests from code and uses that to generate the hyperconverged olm manifest. 

For example, look at the OLM manifest in KubeVirt in that pr I referenced. https://github.com/kubevirt/kubevirt/blob/66abe1ad92614c066412d056b508478c15de03a3/manifests/release/kubevirt.VERSION.csv.yaml.in

You'll see things like ```{{.OperatorDeploymentSpec}}``` sprinkled into that file. The manifest templator is injecting the kubevirt operator deployment spec into that template file, and the deployment spec is actually generated by golang code.

In order to leverage all this work in KubeVirt, this package needs to be following a similar pattern.